tests/test_package_client.py

```diff
@@ -97,35 +91,108 @@ def test_install_configure_start(self, setup_tester_ssh_connection, mender_dist_
     def test_remove_stop(self, setup_tester_ssh_connection, mender_dist_packages_versions):
         result = setup_tester_ssh_connection.run('sudo dpkg -r mender-client')
         assert "Removing mender-client (" + mender_dist_packages_versions["mender-client"] + ")" in result.stdout
 
-        # Check files and directories
-        setup_tester_ssh_connection.run('test ! -f /usr/bin/mender')
-        setup_tester_ssh_connection.run('test ! -e /usr/share/mender')
-        setup_tester_ssh_connection.run('test ! -f /lib/systemd/system/mender-client.service')
-        setup_tester_ssh_connection.run('test ! -f /etc/systemd/system/multi-user.target.wants/mender-client.service')
-        setup_tester_ssh_connection.run('test -d /etc/mender')
-        setup_tester_ssh_connection.run('test -f /etc/mender/mender.conf')
-        setup_tester_ssh_connection.run('test -d /var/lib/mender/')
-        setup_tester_ssh_connection.run('test -f /var/lib/mender/device_type')
-
-        result = setup_tester_ssh_connection.run('sudo journalctl -u mender-client --no-pager')
-        assert "Stopping Mender OTA update service..." in result.stdout
-        assert "Stopped Mender OTA update service." in result.stdout
+        self.check_removed_files(setup_tester_ssh_connection, purge=False)
+
+    @pytest.mark.usefixtures("setup_test_container")
+    def test_purge(self, setup_tester_ssh_connection, mender_dist_packages_versions):
+        result = setup_tester_ssh_connection.run('sudo dpkg -P mender-client')
+        assert "Purging configuration files for mender-client (" + mender_dist_packages_versions["mender-client"] + ")" in result.stdout
+
+        self.check_removed_files(setup_tester_ssh_connection, purge=True)
+
+class TestPackageMenderClientInteractive(PackageMenderClientChecker):
+    """Tests instalation, setup, start, removal and purge of mender-client deb package with
+    in interactive method (i.e. user navigates wizard via stdin).
+    """
+
+    @pytest.mark.usefixtures("setup_test_container")
+    def test_install_configure_start(self, setup_tester_ssh_connection, mender_dist_packages_versions, mender_version):
+        result = setup_tester_ssh_connection.run('uname -a')
+        assert "raspberrypi" in result.stdout
+
+        upload_deb_package(setup_tester_ssh_connection, mender_dist_packages_versions["mender-client"])
+
+        # Install the package using the following flow in the wizard:
+        # Device type (for example raspberrypi3-raspbian): rasbperrytest
+        # Confirm devicetype as rasbperrytest? [Y/n] y
+        # Are you connecting this device to Hosted Mender? [Y/n] n
+        # Do you want to run the client in demo mode? [Y/n] y
+        # Set the IP of the Mender Server: [127.0.0.1] 1.2.3.4
+        result = setup_tester_ssh_connection.run('sudo dpkg -i ' + package_filename(mender_dist_packages_versions["mender-client"]) + """ <<STDIN
+rasbperrytest
+y
+n
+y
+1.2.3.4
+STDIN""")
```

Comment:
Should the resulting values in `/etc/mender/mender.conf` be checked?

Reply:
There are unit tests for this already in `mender` code base that check the actual contents.

Not having an easy way to launch builds on these repos with two PRs, I think that testing here too much will eventually become problematic.

I will add a basic check for IP addr and server url, but not the full file.